agent/runtime-attach/src/main/java/com/microsoft/applicationinsights/attach/ApplicationInsights.java

```diff
@@ -53,8 +56,9 @@ public static void attach() {
 
   private static Optional<String> findJsonConfig() {
 
+    String fileName = findJsonConfigFile();
     InputStream configContentAsInputStream =
-        ApplicationInsights.class.getResourceAsStream("/applicationinsights.json");
+        ApplicationInsights.class.getResourceAsStream("/" + fileName);
```

Comment:
We could, but we would have to add the following kind of code to the current test:

```java
  public static void main(String[] args) {
    System.setProperty("applicationinsights.runtime-attach.configuration.file", "applicationinsights-test.json");
    ApplicationInsights.attach();
    SpringApplication.run(SpringBootApp.class, args);
  }
```

The users would instead add a system property with `-Dapplicationinsights.runtime-attach.configuration.file=applicationinsights-test.json`. So, the test would not represent the typical way of configuring the file name. And so, the test would not document the standard way of configuring the file name.

Another option could be to rewrite a test to be able to add `-Dapplicationinsights.runtime-attach.configuration.file=applicationinsights-test.json`. However, this would require a significant time for an added code pretty simple.

Reply:
maybe add a todo for now? it's not required for this PR.

Reply:
is there some other form of test coverage that may be simpler (but still useful)?

Reply:
An approach could be to change the visibility of some methods to be able to test them. However, I don't recommend doing this. The major drawback of this approach is that we are testing implementation details, not a user use case. So, since the test is tied to implementation details, refactoring (changing the code without changing the behavior) could break it (because we remove a method, for example), adding a cost to test maintenance and slowing down and discouraging future refactoring.